### PR TITLE
First commit of TwinMux unpacker

### PR DESCRIPTION
**Pull request by jpazzini**
First commit of TwinMux unpacker into "EventFilter/TwinMuxRawToDigi/"

### CODE REVIEW

**Comment by davidlange6** [comment]
@jpazzini , @mulhearn - is this package expected to grow? Otherwise, the existing L1 unpacker packages are a better place

**Comment by mulhearn** [comment]
@jpazzini if you prefer, you can PR into https://github.com/cms-l1t-offline/cmssw so that we can integrate there.
@davidlange6 the EventFilter/L1TRawToDigi is reserved for the many subsystems that are all sharing a common unpacker framework.  It would be weird and confusing I think to stuff additional stuff unrelated to the framework there.  But if you want to avoid proliferating packages I'm open to other suggestions...

**Comment by jpazzini** [comment]
@davidlange6 , @mulhearn - currently the package provides an interface to unpack the DT info only, similarly to what is done in the 'legacy' DT trigger with the EventFilter/DTTFRawToDigi; as far as I know both the classes should be kept until the switch to TwinMux is fully integrated. Concerting the future expectations for the package, it will be extended to unpack the RPC informations too, as an interface to the '(DT+RPC)TrackFinder'.

**Comment by davidlange6** [comment]
@mulhearn - lets discuss this week - there are 6 L1 related packages in EventFilter, so I'm reluctant to add another one for ~3 classes.

**Comment by mulhearn** [comment]
Indeed, but it really isn't fair to add legacy L1 to my tab ;-)

**Comment by jpazzini** [comment]
@davidlange6 , @mulhearn - should I then PR into https://github.com/cms-l1t-offline/cmssw or should I wait for an official decision? Thanks.

**Comment by mulhearn** [comment]
@jpazzini let's follow up on this here:

   https://github.com/cms-l1t-offline/cmssw/issues/102

**Comment by davidlange6** [comment]
is this PR now obsolete?

**Comment by mulhearn** [comment]
I think we still need to discuss where to put it.  We have several new unpackers that are not part of the framework in L1TRawToDigi.  I think it would be confusing for future readers to have non-framework unpackers in the same package.  Maybe we should put the rest all into a single package "L1TXRawToDigi"?

**Comment by mulhearn** [comment]
And yes this PR is obsolete, I'll combine with the other second pass unpackers into another PR.  @jpazzini you don't need to worry about any of the name change, etc, I have automated scripts once we decide where the remaining unpackers are going.

File: EventFilter/TwinMuxRawToDigi/src/TwinMuxFEDReader.cc [hunk 1]
```diff
@@ -0,0 +1,688 @@
...
+  if ( amcsecmap_.size() != wheels_.size() )
+    throw cms::Exception("TwinMux_unpacker") << "Configuration file error. Size of \'wheels\' and \'amcsecmap\' differs.\n";  
+  
+  for (size_t wh_i = 0; wh_i < amcsecmap_.size(); ++wh_i){
+    std::map< short, short > whmap;      
+    for (size_t amc_i = 1; amc_i < 13; ++amc_i ){
+      short shift = (12-amc_i)*4;
+      whmap[amc_i] = ( amcsecmap_.at(wh_i) >> shift ) & 0xF;
+    }
+    amcsec_.push_back(whmap);
```
Comment (davidlange6) [inline EventFilter/TwinMuxRawToDigi/src/TwinMuxFEDReader.cc]: using a map to store keys from 1 to 12 is not appropriate. Please use and std::array<short, 12> instead. Or is something else done with this map elsewhere?

**Comment by davidlange6** [comment]
whoops -i commented on the obsolete pr - i guess the comment still will hold.. @mulhearn

**Comment by mulhearn** [comment]
@jpazzini, could you implement David's suggestion, confirm it has no effect, and update this PR?

**Comment by mulhearn** [comment]
PRs on hold, but we can still use this branch while @jpazzini works on this.

**Pull request closed without merging**